esmvaltool/preprocessor/_io.py

```diff
@@ -60,7 +60,7 @@ def load_cubes(files, filename, metadata, constraints=None, callback=None):
     for cube in cubes:
         cube.attributes['_filename'] = filename
         cube.attributes['metadata'] = yaml.safe_dump(metadata)
-        if use_legacy_iris():
+        if not use_legacy_iris():
             # always set fillvalue to 1e+20
             if np.ma.is_masked(cube.data):
```

Comment:
This line is making the data go real way too early in the preprocessor chain. We need to move it

Reply:
Looking more close this was introduced in commit https://github.com/ESMValGroup/ESMValTool/commit/a19daa2acd99abde0db9bbbbd7f27009c26d39db in the desire to ensure the fill_value of the incoming data is as globally defined. @valeriupredoi can comment on this?

However, this hotfix doesn't try to solve this problem in general but wants just to fix that when the "default" iris < 2.0.0 is used nobody gets into this if clause.

Reply:
But that commit was long ago, if that's really causing such problems we would have already noticed.

 @schlunma, are you sure this is the line that causes the performance issue?